### PR TITLE
TRC-16 - chore: add `Storybook`

### PR DESCRIPTION
**Pull request by mmcbride2929**
# TRC-16 - chore: add `Storybook`

## TL/DR
This story adds `Storybook` as a dev dependency and adds relevant scripts to our root folder's package.json file.

## Overview of Changes
`.gitignore` 
- Adds `storybook` generation files

`package.json`
- Configures `dev` script to run `apps/web+api` simultaneously
- Adds `storybook` script to launch from root.

`.npmrc`
- configure package manager for storybook

`.storybook/main.js`
- Configs `storybook` environment

`.storybook/preview.js`
- Configs `storybook` rendering environment

`apps/web/package.json`
- Initialize and install `storybook` dependencies 
- Adds overrides for `enhance-resolve` - used to fix error that occurred when trying to run `pnpm run storybook`
- Adds `schema-utils` dependency used to fix an error that occurs while running `pnpm run storybook`
- Removes un-needed dependencies

### CODE REVIEW

**Comment by mmcbride2929** [comment]
Happy to get storybook running. If I had to redo this again, I would have made more precise commits.

File: apps/web/package.json [hunk 1]
```diff
@@ -9,21 +9,38 @@
...
+    "clean": "rm -rf build",
+    "storybook": "start-storybook -p 6006 -s public",
+    "build-storybook": "build-storybook -s public"
   },
   "dependencies": {
     "@reduxjs/toolkit": "^1.9.2",
     "@types/node-sass": "^4.11.3",
     "@types/react-redux": "^7.1.25",
+    "ajv": "^8.12.0",
+    "concurrently": "^7.6.0",
```
Comment (wijohnst) [inline apps/web/package.json:21]: What are these dependencies?

File: apps/web/src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+const Button = () => {
```
Comment (wijohnst) [inline apps/web/src/components/Button.tsx:1]: Should have `react` in scope and should be typed.

File: apps/web/src/stories/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  label: string;
+  /**
+   * Optional click handler
+   */
+  onClick?: () => void;
+}
+
+/**
+ * Primary UI component for user interaction
+ */
```
Comment (wijohnst) [inline apps/web/src/stories/Button.tsx:29]: Take note of the syntax for all of these comments. It uses this special character set to prefix each multiline comment:

```
/**
```

This is called JSDocs and it is a special kind of comment that can be referenced from the calling scope of the code and use to automate documentation. In our codebase all component props should be commented with JSDocs, just like this example. 

Additionally, all functions should be documented like this:

```ts
/**
* A simple function that adds its two parameters together
*
* @param { number } num1 - the first number to add
* @param { number } num2 - the second numebr to add
* @returns { number }
*/
const sum = (num1: number, num2: number) : number => num1 + num2;
```
Functions should document all of their params, including name and type as well as their return type. Additionally they should have a brief description of what the function does.

You can add the [JSDoc Extension](https://marketplace.visualstudio.com/items?itemName=lllllllqw.jsdoc) to VSCode and easily generate JSDocs templates:

js_docs

File: package.json [hunk 4]
```diff
@@ -6,7 +6,8 @@
     "dev": "turbo run dev --no-cache --continue",
     "format": "prettier --write \"**/*.{ts,tsx,md}\"",
     "lint": "turbo run lint",
-    "test": "turbo run test"
+    "test": "turbo run test",
+    "storybook": "turbo run storybook "
```
Comment (wijohnst) [inline package.json:10]: Looks good.

**Review by wijohnst — CHANGES_REQUESTED** [review]
Good start. The majority of the files that were added are Storybook boilerplate. Let's remove all of those before merging with `main`.

**Review by wijohnst — APPROVED** [review]
Thank you for taking the time to make those changes. Everything looks great. Happy to approve.
